File: backend-end-user/src/main/java/deti/tqs/g305/handymanservicesapp/model/ServiceContract.java
package deti.tqs.g305.handymanservicesapp.model;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
public class ServiceContract {

    private long id;

    private LocalDateTime date;

    private ProviderService providerService;

    private BusinessService businessService;

    private ServiceStatus status;

    private Client client;

    private int review;
    
    public ServiceContract() {
        
    }

    public ServiceContract(BusinessService businessService, ProviderService providerService, ServiceStatus status, Client client, int review) {
        this.businessService=businessService;
        this.status=status;
        this.client=client;
        this.review=review;
        this.providerService=providerService;
    }
}

File: frontend-end-user/src/app/shared/models/ServiceContract.ts
import {Client} from './Client';
import {ProviderService} from './ProviderService';
import {BusinessService} from './BusinessService';
import {ServiceStatus} from './ServiceStatus';

export interface ServiceContract {
    id?: number
    date?: Date,
    providerService?: ProviderService,
    client?: Client,
    businessService?: BusinessService,
    status?: ServiceStatus,
    review?: number
  }